src/main/java/appeng/items/parts/ItemMultipartRendering.java

```diff
@@ -0,0 +1,33 @@
+package appeng.items.parts;
+
+
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.item.ItemStack;
+
+import appeng.bootstrap.IItemRendering;
+import appeng.bootstrap.ItemRenderingCustomizer;
+
+
+public class ItemMultipartRendering extends ItemRenderingCustomizer
+{
+
+	private final ItemMultiPart item;
+
+	public ItemMultipartRendering( ItemMultiPart item )
+	{
+		this.item = item;
+	}
+
+	@Override
+	public void customize( IItemRendering rendering )
+	{
+		rendering.meshDefinition( this::getItemMeshDefinition );
+	}
+
+	private ModelResourceLocation getItemMeshDefinition( ItemStack is )
+	{
+		// TODO: Avoid object creation here. This happens every frame
```

Comment:
Oh it's much easier in this case. The inner resource location already comes from a static final field in an enumeration. The only thing this does is convert it to a ModelResourceLocation by extending it with null. In my opinion that could already be done in the enumeration and kept in there as a static final field as well.